internal/services/integrationtesting/cert_test.go

```diff
@@ -166,17 +170,15 @@ func TestCertRotation(t *testing.T) {
 	)
 
 	require.NoError(t, err)
-	t.Cleanup(func() {
+	defer func() {
```

Comment:
why this change?

Reply:
because I added `<-wait` to make sure the `RunnableServer.Run()` call returned, which can only happen if cancellation is invoked. 
Same for `conn.Close`: if it was happening in `Cleanup` the call to goleak will happen first, and fail